### PR TITLE
Server - Re-enable Lizmap expression filtering on server, follow up #321

### PR DESCRIPTION
**Pull request by Gustry**
* **Funded by**: 3Liz
* **Description**: 
  * Server - Re-enable Lizmap expression filtering on server, follow up #321

If I got it correct @rldhont

### CODE REVIEW

File: lizmap/server/lizmap_accesscontrol.py [hunk 1]
```diff
@@ -34,14 +34,20 @@ def __init__(self, server_iface: 'QgsServerInterface') -> None:
...
         """ Return an additional expression filter """
-        # Disable lizmap layer filter expression while fixing an issue in QGIS Server
-        return ''
-        # QgsMessageLog.logMessage("Lizmap layerFilterExpression", "lizmap", Qgis.Info)
-        # filter_exp = self.get_lizmap_layer_filter(layer)
-        # if filter_exp:
-        #     return filter_exp
-        #
-        # return super().layerFilterExpression(layer)
+        # Disabling Lizmap layer filter expression for QGIS Server < 3.10.13 and < 3.16.1
```
Comment (mdouchin) [inline lizmap/server/lizmap_accesscontrol.py:37]: Could you add also a reference to the PR @rldhont did to fix the issue ?

**Review by mdouchin — APPROVED** [review]

**Comment by Gustry** [comment]
Waiting for the merge upstream.
It might be QGIS 3.16.3 instead of 3.16.2

**Comment by rldhont** [comment]
* Backport 3.16 https://github.com/qgis/QGIS/pull/40585
* Backport 3.10 https://github.com/qgis/QGIS/pull/40584

**Comment by rldhont** [comment]
The fix will be avalaible in 3.18.0, 3.16.2 and 3.10.13

**Review by rldhont — APPROVED** [review]

**Pull request merged**